src/horust/healthcheck.rs

```diff
@@ -157,56 +157,52 @@ fn client_mock() -> bool {
         }
     }
 }
-
-fn start_server() {
-    let listener = TcpListener::bind("127.0.0.1:9999").unwrap();
-    listener
-        .set_nonblocking(true)
-        .expect("Cannot set to non-blocking");
-    // accept connections and process them, spawning a new thread for each one
-    println!("Server listening on port 9999");
-    for stream in listener.incoming() {
-        match stream {
-            Ok(stream) => {
-                // connection succeeded
-                handle_test_connection(stream);
-            }
-            Err(e) if e.kind() == ErrorKind::WouldBlock => {
-                continue;
-                /* connection failed */
-            }
-            Err(_) => {}
-        }
-    }
-    // close the socket server
-    drop(listener);
-}
-
-fn handle_test_connection(mut stream: TcpStream) {
-    let mut buffer = [0; 512];
-
-    stream.read(&mut buffer).unwrap();
-
-    let response = "HTTP/1.1 200 OK\r\n\r\n";
-
-    stream.write(response.as_bytes()).unwrap();
-    stream.flush().unwrap();
-}
+ */
 
 #[cfg(test)]
 mod test {
     use crate::horust::error::Result;
     use crate::horust::formats::{Event, Healthiness, Service, ServiceName, ServiceStatus};
     use crate::horust::healthcheck;
-    // use crate::horust::healthcheck::check_http_endpoint;
+    use crate::horust::healthcheck::check_http_endpoint;
     use crate::horust::healthcheck::healthchecks;
-    use crate::horust::healthcheck::{client_mock, start_server};
-    use rand::Rng;
     use std::collections::HashMap;
-    use std::net::{Ipv4Addr, SocketAddrV4};
+    use std::io::prelude::*;
+    use std::io::ErrorKind;
+    use std::net::TcpListener;
+    use std::net::TcpStream;
     use std::thread;
     use tempdir::TempDir;
-
+    fn start_server() {
+        let listener = TcpListener::bind("127.0.0.1:9999").unwrap();
+        listener
+            .set_nonblocking(true)
+            .expect("Cannot set to non-blocking");
+        // accept connections and process them, spawning a new thread for each one
+        println!("Server listening on port 9999");
+        for stream in listener.incoming() {
+            match stream {
+                Ok(stream) => {
+                    // connection succeeded
+                    handle_test_connection(stream);
+                }
+                Err(e) if e.kind() == ErrorKind::WouldBlock => {
+                    continue;
+                    /* connection failed */
+                }
+                Err(_) => {}
+            }
+        }
+        // close the socket server
+        drop(listener);
```

Comment:
Is this going to work? I'm not sure. What you had before was fine, inside the stream, after you handle_test_connection then you need to break the for loop. Otherwise you could also just remove the for and wait for a single connection (not sure how, it might be enough to do `incoming().into_iter().take(1).collect()`. But you might want to search for it).
Usually you don't need to explicitly drop anything in rust, it will be automagically dropped as soon as you exit the function.

Reply:
The thing is that incoming() will be blocking. so the sibling code under it will not run until incoming is done. And we are sending the client request after the server is started.

Reply:
Not sure what you mean with "sibiling" code. The client request runs in another thread right? So what's the matter with that?